calblueprint/meepanyar
# [Translation Support] Modify hook to support variable input

# In this PR 
This PR addresses [the pass-by-value challenge](https://www.notion.so/calblueprint/Translation-Support-Design-Document-7703018a8c4e403dab8aaa49d2a322e7#f545439216a24965bba433d722fd6648) of our [roadmap for translation support](https://www.notion.so/calblueprint/Translation-Support-Design-Document-7703018a8c4e403dab8aaa49d2a322e7) by adding an optional argument, `input`, which will be used to handle variable inputs.

The important thing to note about this PR is that it relies [these DB constraints](https://www.notion.so/calblueprint/Translation-Support-Design-Document-7703018a8c4e403dab8aaa49d2a322e7#e2bec25623f24f29b3fa26bb30049bd7) in the translation DB. 

## How to review
1. Choose any frontend screen or component with some variable input. In this example, we will use `CustomerCard.tsx`. 
2. Import and initialize the translation hook (same as #57) with `import { useInternationalization } from '../lib/i18next/translator';` at the top of the file and `const intl = useInternationalization();` in the component. 
3. Let's localize the text, `Last Updated: {props.date}` be replacing that line with the following (line 107 atow).
```
<Typography className={classes.updatedText}>{intl("Last Updated [Date]", props.date)}</Typography>
```
4. If you want to test this in Burmese, change your browser language to Burmese or modify `device.ts#detectLanguage` to return "my".
5. Run `npm start` and you should see the changes reflected on the customer cards `http://localhost:3000/customers`. Here's one example: 

![Screen Shot 2021-04-07 at 8 42 56 PM](https://user-images.githubusercontent.com/44735047/113965256-e330e400-97e1-11eb-9b3b-efc3a9dbdb4f.png)

**Note**: Some customer cards will not have enough vertical padding to display the full Burmese text. This is out of scope for this PR, but I thought I would point it out. 


## Relevant Links
- [Design document for translation support](https://www.notion.so/calblueprint/Translation-Support-Design-Document-7703018a8c4e403dab8aaa49d2a322e7) 

### Related PRs
- #57 
- #61 
- #76  

## Next steps
- Propagating this hook throughout the entire app (essentially comes down to replacing any hardcoded strings with our new `useIntl` hook). 

CC: @julianrkung
